dataactbroker/handlers/fileHandler.py

```diff
@@ -479,9 +498,134 @@ def checkD1File(self, submission_id=None):
 
         return JsonResponse.create(StatusCode.OK, response)
 
+    def loadGenerateRequest(self):
+        """ Pull information out of request object and return it
+
+        Returns: tuple of submission ID and file type
+
+        """
+        requestDict = RequestDictionary(self.request)
+        if not (requestDict.exists("submission_id") and requestDict.exists("file_type")):
+            exc = ResponseException("Generate file route requires submission_id and file_type",
+                                    StatusCode.CLIENT_ERROR)
+            return JsonResponse.error(exc, exc.status)
+
+        submission_id = requestDict.getValue("submission_id")
+        file_type = requestDict.getValue("file_type")
+        return submission_id, file_type
+
+    def generateFile(self):
+        """ Start a file generation job for the specified file type """
+        submission_id, file_type = self.loadGenerateRequest()
+        job = self.interfaces.jobDb.getJobBySubmissionFileTypeAndJobType(submission_id, self.EXTERNAL_FILE_TYPE_MAP[file_type], "file_upload")
+        # Check prerequisites on upload job
+        if not self.interfaces.jobDb.runChecks(job.job_id):
+            exc = ResponseException("Must wait for completion of prerequisite validation job", StatusCode.CLIENT_ERROR)
+            return JsonResponse.error(exc, exc.status)
+        if file_type == "D1":
+            # TODO could this function return S3 file location
+            result = self.generateD1File()
```

Comment:
Right, this is in an awkward halfway state where generateD1File and generateD2File can be called directly from the old routes.  The next PR will be removing those routes and consolidating the generateD1File and generateD2File functions into a more generic function that handles E and F as well.  The consolidated function will certainly take submission ID and file type as arguments, probably the job as well depending on what it returns.